src/messages/mod.rs

```diff
@@ -219,16 +277,17 @@ impl SignedRoutingMessage {
             SecurityMetadata::Full(ref security_metadata) => {
                 chain.check_trust(security_metadata.proof_chain())
             }
-            SecurityMetadata::None => true,
+            SecurityMetadata::None | SecurityMetadata::Single(_) => true,
             SecurityMetadata::Partial(_) => false,
         }
     }
 
     /// Returns the security metadata validating the message.
     pub fn source_section_key_info(&self) -> Option<&SectionKeyInfo> {
         match self.security_metadata {
-            SecurityMetadata::None => None,
-            SecurityMetadata::Partial(_) => None,
+            SecurityMetadata::None | SecurityMetadata::Partial(_) | SecurityMetadata::Single(_) => {
```

Comment:
I thought even it is a `single`, it's still a security metadata from source ?

Reply:
We do not have a mechanism to tie the BLS SectionKey to the specific ED key of a member.
This would be what this https://github.com/maidsafe/routing/issues/1751 is about, but I don't think we can really do it for now (since we still use Parsec signatures for SectionInfo) until swtiching to real BLS (Whay I mention in the commit comment).